tests/test_post_gen_project.py

```diff
@@ -9,40 +10,105 @@
     get_repo_dir,
 )
 
+@pytest.mark.parametrize("repos",[
+    # test get_repo_dir when "scikit-package-manuscript" repo exists.
+    # C1: with only "scikit-package-manuscript" repo
+    ("scikit-package-manuscript",),
+    # C2: with "scikit-package-manuscript" and other repos
+    ("scikit-package-manuscript", "foo", "bar"),
+]
+)
+def test_get_repo_dir(user_filesystem, repos):
+    home_path, ck_path, cwd_path = user_filesystem
+    skm_repo_path = home_path / ".cookiecutters" / "scikit-package-manuscript"
+    os.environ["HOME"] = str(home_path)
+    for repo in repos:
+        repo_path = ck_path / repo
+        repo_path.mkdir()
+    actual = get_repo_dir()
+    expected = skm_repo_path
+    assert expected == actual
 
-def test_get_repo_dir(monkeypatch, user_filesystem):
-    home_dir = user_filesystem[0]
-    monkeypatch.setenv("HOME", str(home_dir))
-    repo_path = home_dir / ".cookiecutters" / "scikit-package-manuscript"
 
-    assert repo_path == get_repo_dir()
+@pytest.mark.parametrize("repos",[
+    # test get_repo_dir when "scikit-package-manuscript" repo does not exist.
+    # C1: empty ".cookiecutters"
+    (),
+    # C2: without "scikit-package-manuscript"
+    ("foo", "bar"),
+]
+)
+def test_get_repo_dir_bad(user_filesystem, repos):
+    home_path, ck_path, cwd_path = user_filesystem
+    os.environ["HOME"] = str(home_path)
+    for repo in repos:
+        repo_path = ck_path / repo
+        repo_path.mkdir()
+    with pytest.raises(FileNotFoundError, match=r"Couldn't find scikit-package-manuscript,") as exc_info:
+        get_repo_dir()
 
+@pytest.mark.parametrize("existing_templates, template",[
+    # test if copy_journal_template_files can find the expected template.
+    # C1: with only one template
+    (["article"], "article"),
+    # C2: with multiple templates
+    (["article", "foo", "bar"], "article")
+]
+)
+def test_copy_journal_template_files_find_template(user_filesystem_with_repo, existing_templates, template):
+    home_path, skm_repo_path, cwd_path = user_filesystem_with_repo
+    os.environ["HOME"] = str(home_path)
+    for existing_template in existing_templates:
+        existing_template_path = skm_repo_path / "templates" / existing_template
+        existing_template_path.mkdir(parents=True, exist_ok=True)
+    
+    cwd_path = copy_journal_template_files(template, cwd_path)
+
```

Comment:
is there a missing assert here?  Did you check that this test fails when the function is not correct?  Always make sure a test fails before it passes.